Updated DesiMock.py

ZechangSun opened the pull request:
Dear Professor Prochaska, this is the lastest DesiMock.py, which has been changed according to your advice.

samwang141224 committed:
Update Sightline.py

zoujiaqi99 committed:
Update Sightline.py

zoujiaqi99 committed:
Update Sightline.py

ZechangSun committed:
DesiMock.py

Dear Professor Prochaska, this is the updated DesiMock.py, according to your advice.

profxj commented on dla_cnn/data_model/DesiMock.py:
```diff
@@ -0,0 +1,88 @@
+    :attribute data, dict, using each spectra's id as its key and a dict of all data we need of this spectra as its value
+      its format like spectra_id: {'FLUX':flux,'ERROR':error,'z_qso':z_qso, 'RA':ra, 'DEC':dec, 'DLAS':dlas information} ,
+      dla information is a tuple, and its format is like (dla_id, z_qso, NHI)
+    :attribute split_point_br, the length of the flux_b,the split point of  b channel data and r channel data
```
Please indicate this is an `int`

ZechangSun replied:
Sorry for not checking in time, I have updated it.

profxj commented on dla_cnn/data_model/DesiMock.py:
```diff
@@ -0,0 +1,88 @@
+    :attribute wavelength array-like, the wavelength of all spectrum (all spectrum share same wavelength array)
+    :attribute data, dict, using each spectra's id as its key and a dict of all data we need of this spectra as its value
+      its format like spectra_id: {'FLUX':flux,'ERROR':error,'z_qso':z_qso, 'RA':ra, 'DEC':dec, 'DLAS':dlas information} ,
+      dla information is a tuple, and its format is like (dla_id, z_qso, NHI)
```
What if there is more than 1 DLA on the sightline?
Is it then a series of tuples?

profxj replied:
And actually, it is more than DLAs.  We are taking all systems with
NHI > 19.3.

ZechangSun replied:
Yes, it will be a series of tuples. I have added some comments about it.

profxj commented on dla_cnn/data_model/DesiMock.py:
```diff
@@ -0,0 +1,88 @@
+    def read_fits_file(self, spec_path, truth_path, zbest_path):
+        """
+        read Desi Mock spectrum from a fits file, load all spectrum as a DesiMock object
+        :param spec_path: spectrum file path
```
Indicate that each of these paths is a `str`

ZechangSun replied:
Have done.

profxj commented on dla_cnn/data_model/DesiMock.py:
```diff
@@ -0,0 +1,88 @@
+        truth = fits.open(truth_path)
+        zbest = fits.open(zbest_path)
+
+        #spec[2].data ,spec[7].data and spec[12].data are the wavelength data
```
add:  for the b, r, and z cameras

ZechangSun replied:
Have done. Thanks!

profxj commented on dla_cnn/data_model/DesiMock.py:
```diff
@@ -0,0 +1,88 @@
+
+        self.data = {spec_id[i]:{'FLUX':flux[i],'ERROR': error[i], 'z_qso':z_qso[i] , 'RA': ra[i], 'DEC':dec[i], 'DLAS':spec_dlas[spec_id[i]]} for i in range(len(spec_id))}
+
+        return self.wavelength,self.data,self.split_point_br,self.split_point_rz,self.data_size
```
There is no need to return these items.  They are all in the class.

ZechangSun replied:
I have changed it. Thanks for your advice!

profxj commented on dla_cnn/data_model/DesiMock.py:
```diff
@@ -0,0 +1,88 @@
+
+        return self.wavelength,self.data,self.split_point_br,self.split_point_rz,self.data_size
+
+    def get_sightline(self,id):
```
Modify this method to allow the user to grab only 1 of the 3 cameras.
It can default to all 3.  e.g.

`def get_sightline(self,id,camera='all'):`

then

`if camera == 'b':`
  Load up only the wavelength and data for the blue camera.

ZechangSun replied:
I have changed `get_sightline(self,id,camera='all')` to allow the user to grab the camera they want.  Thanks for your advice!

profxj commented on dla_cnn/data_model/Sightline.py:
```diff
@@ -3,24 +3,26 @@
             flux (np.ndarray):
-            loglam (np.ndarray):
-                log10 of observed wavelength values
+            wavelength (np.ndarray):
```
Revert back to `loglam` instead of `wavelength`

ZechangSun replied:
I have changed it back.

profxj commented on dla_cnn/data_model/Sightline.py:
```diff
@@ -3,24 +3,26 @@
         self.dlas = dlas
         self.z_qso = z_qso
-        self.data_markers = []     # An array of DataMarker objects which mark areas of the sightline
+        self.error = error    # error = 1./ np.sqrt(ivar)
```
I forget what `data_markers` are but I think you should
not remove them.

ZechangSun replied:
OK, I have added it.

profxj commented:
Also, please create a Notebook in a desi/nb/ folder
showing how to use this code.

Have it load a `Sightline` of only the blue camera and
plot it.  Ideally one with a DLA.

profxj commented on dla_cnn/data_model/DesiMock.py:
```diff
@@ -0,0 +1,88 @@
+        sightline.z_qso = self.data[id]['z_qso']
+        sightline.ra = self.data[id]['RA']
+        sightline.dec = self.data[id]['DEC']
+        sightline.dlas = self.data[id]['DLAS']
```
This needs to be a list of `Dla` objects as defined in data_model/Dla.py

ZechangSun replied:
Should I add those attributes to the Dla class since the Dla class only have entral_wavelength and col_density?

profxj replied:
yes, it is fine to add more attributes.

ZechangSun replied:
Ok, I have updated it. Thanks for your advice!

ZechangSun committed:
Update DesiMock.py

ZechangSun committed:
Update Sightline.py

ZechangSun committed:
Update Dla.py

ZechangSun committed:
Update DesiMock.py

ZechangSun committed:
Update Dla.py

ZechangSun committed:
Update DesiMock.py

profxj commented:
Ok @ZechangSun -- Almost done.  Two more things:

1.  Please move DesiMock.py into the desi/ folder
2.  Please generate a simple Notebook that shows an example of using the code including making a plot of a spectrum with a DLA.  Put that Notebook in desi/nb

Thanks!

profxj commented on dla_cnn/data_model/DesiMock.py:
```diff
@@ -0,0 +1,98 @@
+            sightline.z_qso = self.data[id]['z_qso']
+            sightline.ra = self.data[id]['RA']
+            sightline.dec = self.data[id]['DEC']
+            sightline.dlas = self.data[id]['DLAS'][start_point:end_point]
```
This looks like a bug.  That is the DLAS don't know about the camera.

ZechangSun replied:
Yes, it is. I have corrected it and tested the code. Thanks!
Besides, I have moved it into the desi/folder and uploaded a jupyter notebook to desi/nb/folder to show its usage.Sorry for delaying such a long time, I just want to update the code to the final version before showing its usage.

profxj replied:
Thanks for the Notebook!

Please remove all the cells before [30]
and confirm things still work.

You will need to add this to cell [31]

`from dla_cnn.desi.DesiMock import DesiMock`

ZechangSun replied:
Ok, I have changed it. Thanks!

profxj replied:
Almost.  Your input statement doesn't include `dla_cnn`
Please update that.

ZechangSun replied:
Yes, I only install the desi module in my computer, so this may cause some confusing. Now I have updated both the DesiMock.py and the notebook. Thanks for your advice.

ZechangSun committed:
Update DesiMock.py

ZechangSun committed:
Create DesiMock.py

ZechangSun committed:
Create DesiMock.ipynb

ZechangSun committed:
Delete DesiMock.ipynb

ZechangSun committed:
Create nb

ZechangSun committed:
Delete nb

ZechangSun committed:
Create desimock.py

ZechangSun committed:
Add files via upload

ZechangSun committed:
Delete Usage of DesiMock.ipynb

ZechangSun committed:
Add files via upload

ZechangSun committed:
Delete desimock.py

samwang141224 committed:
Add files via upload

ZechangSun committed:
Update DesiMock.py

ZechangSun committed:
Delete DesiMock.py

ZechangSun committed:
Update Dla.py

profxj commented on dla_cnn/desi/model_v2.1.py:
```diff
@@ -0,0 +1,243 @@
+""" TensorFlow Models for DLA finder"""
```
While it is great to see this code, please take it out of
this PR, put it in its own branch and then submit a PR 
for it when it is ready.

profxj replied:
While it is great to see this code, please take it out of
this PR, put it in its own branch and then submit a PR 
for it when it is ready.

ZechangSun committed:
Update DesiMock.py

ZechangSun committed:
Create desimock.py

ZechangSun committed:
Delete Usage of DesiMock.ipynb

ZechangSun committed:
Add files via upload

ZechangSun committed:
Delete desimock.py

samwang141224 committed:
Delete model_v2.1.py

I will delete this code in read_mock_spectra branch and keep it in model_v2 branch

profxj commented on dla_cnn/desi/DesiMock.py:
```diff
@@ -1,5 +1,7 @@
 from astropy.io import fits
 import numpy as np
+from data_model.Sightline import Sightline
```
This should be

`from dla_cnn.data_model.Sightline import Sightline`

and the same for `Dla`

ZechangSun replied:
Ok!I will change it.

ZechangSun committed:
Update DesiMock.py

ZechangSun committed:
Update Usage of DesiMock.ipynb

profxj reviewed (approved):
All set!

Great job!

profxj commented:
merging
